test/profile.test.js

```diff
@@ -46,20 +72,54 @@ test('threads.profile gives threads for lucy not mary', (t) => {
 
     pull.collect((err, threads) => {
       t.error(err);
-      t.equals(threads.length, 1, 'only one thread');
-      const thread = threads[0];
-      t.equals(thread.full, true, 'thread comes back full');
-      t.equals(thread.messages.length, 2, 'thread has 2 messages');
+      t.equals(threads.length, 2, 'two threads');
+
+      // Mary's thred
+      t.equals(threads[0].full, true, 'thread comes back full');
+      t.equals(threads[0].messages.length, 2, 'thread has 2 messages');
+
+      t.deepEquals(
+        threads[0].messages.map(m => m.value.content),
+        [
+          { type: 'post', text: 'Root from mary' },
+          { type: 'post', text: 'Reply from lucy (2)', root: threads[0].messages[0].key },
+        ]
+      )
+      
+      // Lucy's thread
+      t.equals(threads[1].full, true, 'thread comes back full');
+      t.equals(threads[1].messages.length, 2, 'thread has 2 messages');
+
+      t.deepEquals(
+        threads[1].messages.map(m => m.value.content),
+        [
+          { type: 'post', text: 'Root from lucy' },
+          { type: 'post', text: 'Reply from lucy (1)', root: threads[1].messages[0].key },
+        ]
+      )
+
+      t.comment('> opts.initiatedOnly') 
+      pull(
+        ssb.threads.profile({ id: lucyKeys.id, initiatedOnly: true }),
+        pull.collect((err, threads) => {
+          t.error(err);
+          t.equals(threads.length, 1, '1 thread');
 
-      const msgs = thread.messages;
-      const rootKey = msgs[0].key;
-      t.equals(msgs[0].value.content.root, undefined, '1st message is root');
-      t.equals(msgs[0].value.content.text, 'Root from lucy');
+          // Lucy's thread
+          t.equals(threads[0].full, true, 'thread comes back full');
+          t.equals(threads[0].messages.length, 2, 'thread has 2 messages');
 
-      t.equals(msgs[1].value.content.root, rootKey, '2nd message is not root');
-      t.equals(msgs[1].value.content.text, 'Reply from lucy');
+          t.deepEquals(
+            threads[0].messages.map(m => m.value.content),
+            [
+              { type: 'post', text: 'Root from lucy' },
+              { type: 'post', text: 'Reply from lucy (1)', root: threads[0].messages[0].key },
+            ]
+          )
 
-      ssb.close(t.end);
+          ssb.close(t.end);
+        })
+      )
     }),
   );
 });
```

Comment:
Sorry, you're going to have to redo your changes to this file. Please don't modify any existing tests (i.e. revert all of this diff in this file). If there is no breaking change, then the existing tests should pass. Just add one new test where we use `opts.initiatedOnly = true`.